arangod/Aql/TraversalNode.cpp

```diff
@@ -544,6 +550,195 @@ void TraversalNode::doToVelocyPack(VPackBuilder& nodes, unsigned flags) const {
+        aql::Condition::collectOverlappingMembers(
+            _plan, options()->tmpVar(), remainderCondition, indexCondition,
+            toRemove, nullptr, false);
+        size_t n = remainderCondition->numMembers();
+
+        std::unique_ptr<aql::Expression> expression = nullptr;
+
+        if (n != toRemove.size()) {
+          // Slow path need to explicitly remove nodes.
+          for (; n > 0; --n) {
+            // Now n is one more than the idx we actually check
+            if (toRemove.find(n - 1) != toRemove.end()) {
+              // This index has to be removed.
+              remainderCondition->removeMemberUnchecked(n - 1);
+            }
+          }
+          expression = std::make_unique<aql::Expression>(_plan->getAst(),
+                                                         remainderCondition);
+        }
+
+        auto container = aql::utils::extractNonConstPartsOfIndexCondition(
+            ast, getRegisterPlan()->varInfo, false, false, indexCondition,
+            options()->tmpVar());
+
+        indexAccessors.emplace_back(indexToUse, indexCondition, memberToUpdate,
+                                    std::move(expression), std::move(container),
+                                    i);
+        break;
+      }
+      case TRI_EDGE_OUT: {
+        std::shared_ptr<Index> indexToUse{nullptr};
+        aql::AstNode* indexCondition =
+            conditionBuilder.getOutboundCondition()->clone(ast);
+        bool res = aql::utils::getBestIndexHandleForFilterCondition(
+            *_edgeColls[i], indexCondition, options()->tmpVar(), 1000,
+            aql::IndexHint(), indexToUse, onlyEdgeIndexes);
+        if (!res) {
+          THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_INTERNAL,
+                                         "expected edge index not found");
+        }
+
+        std::optional<size_t> memberToUpdate{std::nullopt};
+        std::pair<arangodb::aql::Variable const*,
+                  std::vector<basics::AttributeName>>
+            pathCmp;
+        for (size_t x = 0; x < indexCondition->numMembers(); ++x) {
+          // We search through the nary-and and look for EQ - _from/_to
+          auto eq = indexCondition->getMemberUnchecked(x);
+          if (eq->type !=
+              arangodb::aql::AstNodeType::NODE_TYPE_OPERATOR_BINARY_EQ) {
+            // No equality. Skip
+            continue;
+          }
+          TRI_ASSERT(eq->numMembers() == 2);
+          // It is sufficient to only check member one.
+          // We build the condition this way.
+          auto mem = eq->getMemberUnchecked(0);
+          if (mem->isAttributeAccessForVariable(pathCmp, true)) {
+            if (pathCmp.first != _tmpObjVariable) {
+              continue;
+            }
+            if (pathCmp.second.size() == 1 &&
+                pathCmp.second[0].name == StaticStrings::FromString) {
+              memberToUpdate = x;
+              break;
+            }
+            continue;
+          }
```

Comment:
This piece looks similar to what we have in the TRI_EDGE_IN: case besides the `.name  == StaticStrings::FromString`
Can we try to get this as a helper method instead?

Reply:
Moved both into a lambda helper method.